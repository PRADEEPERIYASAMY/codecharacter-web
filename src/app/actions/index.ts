export * from 'app/actions/dashboard';
