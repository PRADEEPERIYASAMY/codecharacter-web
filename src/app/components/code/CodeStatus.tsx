import * as styles from 'app/styles/CodeStatus.module.css';
import * as CodeStatusInterfaces from 'app/types/code/CodeStatus';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';

export class CodeStatus extends React.Component<CodeStatusInterfaces.Props, {}> {
  public render() {
    const { currentCommitHash, width } = this.props;
    return (
      <Row
        className={classnames(styles.CodeStatusRow)}
        style={{
          width,
        }}
      >
        <Col sm={8} className={classnames(styles.CommitStatusCol)}>
          <p className={classnames(styles.StatusText)} style={{ textAlign: 'left' }}>
            {' '}
            On Commit: {currentCommitHash.slice(0, 7)}{' '}
            {currentCommitHash !== 'latest' ? '(Non editable)' : ''}
          </p>
        </Col>
        <Col sm={4} className={classnames(styles.StatusTextCol)}>
          <p className={classnames(styles.StatusText)} style={{ textAlign: 'right' }}>
            {' '}
            Saved 2 min ago{' '}
          </p>
        </Col>
      </Row>
    );
  }
}
