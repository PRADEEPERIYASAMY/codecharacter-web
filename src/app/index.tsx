import { Dashboard } from 'app/components/Dashboard';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
