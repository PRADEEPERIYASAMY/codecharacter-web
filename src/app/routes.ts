export enum Routes {
  ROOT = '/',
  LOGIN = '/login',
  REGISTER = '/register',
  LEADERBOARD = '/leaderboard',
  USER_PROFILE_MODEL = '/profile',
<<<<<<< HEAD
  GITHUB_OAUTH = '/login/github',
  GOOGLE_OAUTH = '/login/google',
  USER_ACTIVATION = '/user-activate',
=======
  PROFILE_USER_STATS = '/profile/:username',
>>>>>>> Add route for /profile/:user and Add the required container
}
