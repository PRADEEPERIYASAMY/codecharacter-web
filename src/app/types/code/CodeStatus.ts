export interface StateProps {
  currentCommitHash: string;
}

export interface OwnProps {
  width: number;
}

export type Props = OwnProps & StateProps & {};
