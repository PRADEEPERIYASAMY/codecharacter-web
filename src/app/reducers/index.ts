import { codeReducer } from 'app/reducers/code/Code';
import { editorReducer } from 'app/reducers/code/Editor';
import { dashboardReducer } from 'app/reducers/Dashboard';
import { gameLogReducer } from 'app/reducers/GameLog';
import { leaderboardReducer } from 'app/reducers/Leaderboard';
import { notificationReducer } from 'app/reducers/Notification';
import { userReducer } from 'app/reducers/User';
import * as CodeInterfaces from 'app/types/code/Code';
import * as EditorInterfaces from 'app/types/code/Editor';
import * as DashboardInterfaces from 'app/types/Dashboard';
import * as GameLogInterfaces from 'app/types/GameLog';
import * as LeaderboardInterfaces from 'app/types/Leaderboard';
import * as UserInterfaces from 'app/types/User';
import { routerReducer, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';

export const rootReducer = combineReducers({
  code: codeReducer,
  dashboard: dashboardReducer,
  editor: editorReducer,
  gameLog: gameLogReducer,
  leaderboard: leaderboardReducer,
  notification: notificationReducer,
  router: routerReducer,
  user: userReducer,
});

export interface RootState {
  code: CodeInterfaces.CodeStoreState;
  dashboard: DashboardInterfaces.DashboardStoreState;
  editor: EditorInterfaces.EditorStoreState;
  leaderboard: LeaderboardInterfaces.LeaderboardStoreState;
  router: RouterState;
  gameLog: GameLogInterfaces.GameLogStoreState;
  user: UserInterfaces.UserStoreState;
}
