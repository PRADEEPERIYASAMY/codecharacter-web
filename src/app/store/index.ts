import { logger } from 'app/middleware';
import { rootReducer } from 'app/reducers';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

export function configureStore(initialState?: object) {
  let middleware = applyMiddleware(logger);

  if (process.env.NODE_ENV !== 'production') {
    middleware = composeWithDevTools(middleware);
  }

  const store = createStore(rootReducer, initialState, middleware);

  if (module.hot) {
    module.hot.accept('app/reducers', () => {
      const nextReducer = require('app/reducers');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
