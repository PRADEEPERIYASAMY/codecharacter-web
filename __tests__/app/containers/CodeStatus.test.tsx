import CodeStatus from 'app/containers/code/CodeStatus';
import { configureStore } from 'app/store';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('CodeStatus Container', () => {
  const { store } = configureStore();
  const wrapper = shallow(<CodeStatus width={250} />, {
    context: {
      store,
    },
  });

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
