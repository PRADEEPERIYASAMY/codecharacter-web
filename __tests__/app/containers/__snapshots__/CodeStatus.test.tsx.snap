// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeStatus Container Should render 1`] = `
<CodeStatus
  currentCommitHash="latest"
  width={250}
/>
`;
